test/functional/provider/ruby_spec.lua

```diff
@@ -0,0 +1,80 @@
+local helpers = require('test.functional.helpers')(after_each)
+
+local eq = helpers.eq
+local feed = helpers.feed
+local clear = helpers.clear
+local funcs = helpers.funcs
+local meths = helpers.meths
+local insert = helpers.insert
+local expect = helpers.expect
+local command = helpers.command
+local write_file = helpers.write_file
+local curbufmeths = helpers.curbufmeths
+
+do
+  clear()
+  command('let g:prog = provider#ruby#Detect()')
+  local prog = meths.get_var('prog')
+
+  if prog == '' then
+    pending(
+    'Couldn\'t find the neovim RubyGem. Install it with `gem install neovim`.',
```

Comment:
The escape can be avoided by using double quotes

    "Couldn't find the neovim RubyGem. Install it with `gem install neovim`.",